### PR TITLE
TechDebt: Pipeline Cleanup/Improvements

### PR DESCRIPTION
**Pull request by NickPhura**
# Overview

## Links to Jira tickets

N/A

## Description of relevant changes

- Misc formatting

- Tightened the times on the startup/readiness/liveness probes
  - Ideally the startup time is improved as they check more often now

- Re-wrote some of the utils inherited from another team
  - Previously they weren't actually waiting correctly, and were harder to decipher, etc.

- Added an extra pod check in the db setup deployment.  It now checks that the database pod is ready before running the migrations, will hopefully eliminate the odd random failure we got when the setup pod would report it couldn't connect to the database.

- Added a makefile command for installing the pipeline packages (just to make it easier)

## PR Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### Code

- [ ] New files/classes/functions have appropriately descriptive names and comment blocks to describe their use/behaviour
- [ ] I have avoided duplicating code when possible, moving re-usable pieces into functions
- [ ] I have avoided hard-coding values where possible and moved any re-usable constants to a constants file
- [ ] My code is as flat as possible (avoids deeply nested if/else blocks, promise chains, etc)
- [ ] My code changes account for null/undefined values and handle errors appropriately
- [ ] My code uses types/interfaces to help describe values/parameters/etc, help ensure type safety, and improve readability

### Style

- [ ] My code follows the established style conventions
- [ ] My code uses native material-ui components/icons/conventions when possible

### Documentation

- [ ] I have commented my code sufficiently, such that an unfamiliar developer could understand my code
- [ ] I have added/updated README's and related documentation, as needed

### Tests

- [ ] I have added/updated unit tests for any code I've added/updated
- [ ] I have added/updated the Postman requests/tests to account for any API endpoints I've added/updated

### Linting/Formatting

- [ ] I have run the linter and fixed any issues, as needed  
       _See the `lint` commands in package.json_
- [ ] I have run the formatter and fixed any issues, as needed  
       _See the `format` commands in package.json_

### SonarCloud

- [ ] I have addressed all SonarCloud Bugs, Vulnerabilities, Security Hotspots, and Code Smells

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -317,6 +317,23 @@ format-fix: ## Runs `npm run format-fix` for all projects
...
 	@cd database && npm run format-fix && cd ..
 
+## ------------------------------------------------------------------------------
+## Run `npm` commands for all projects ./.pipeline
+## ------------------------------------------------------------------------------
+pipeline-install: ## Runs `npm install` for all projects
+	@echo "==============================================="
+	@echo "Running /api/.pipeline install"
+	@echo "==============================================="
+	@cd api/.pipeline && npm install && cd ../..
```
Comment (curtisupshall) [inline Makefile:327]: Thoughts on using `pushd` and `popd` in lieu of of `cd`?

File: app/.pipeline/config.js [hunk 2]
```diff
@@ -71,12 +73,9 @@ const phases = {
     instance: `${name}-dev-${deployChangeId}`,
     version: `${deployChangeId}-${changeId}`,
     tag: `dev-${version}-${deployChangeId}`,
-    host:
-      (isStaticDeployment && (staticUrls.dev)) ||
-      `${name}-${changeId}-a0ec71-dev.apps.silver.devops.gov.bc.ca`,
+    host: (isStaticDeployment && staticUrls.dev) || `${name}-${changeId}-a0ec71-dev.apps.silver.devops.gov.bc.ca`,
```
Comment (curtisupshall) [inline app/.pipeline/config.js:76]: Would a ternary be more readible here? I'm pretty sure it is functionally the same but I'm not certain

Comment (NickPhura) [reply app/.pipeline/config.js:76]: Should be the same in the end

That being said, there is definitely more to clean up in all of these pipeline files, especially around all of these variables, and how they get passed around (its a bit hard to trace them through the steps)

But maybe in another ticket

**Review by curtisupshall — APPROVED** [review]
Looks great! 🧹

File: Makefile (hunk 1 above)
Comment (NickPhura) [reply Makefile:327]: No strong thoughts, wasn't aware of pushd or popd tbh.

Did a quick search, seems like both do more or less the same thing.

I'd probably just leave them as is, unless we have a reason to change them

**Pull request merged**